src/features/Apiexplorer/SubscribeRenderer/index.tsx

```diff
@@ -25,30 +25,30 @@ function SubscribeRenderer<T extends TSocketSubscribableEndpointNames>({
 }: IResponseRendererProps<T>) {
   const { is_logged_in } = useAuthContext();
   const { disableSendRequest } = useDisableSendRequest();
-  const { full_response, is_loading, subscribe, error } = useSubscription<T>(name);
+  const { full_response, is_loading, subscribe, unsubscribe, is_subscribed, error } =
+    useSubscription<T>(name);
   const [response_state, setResponseState] = useState(false);
   const [toggle_modal, setToggleModal] = useState(false);
   const [is_not_valid, setIsNotValid] = useState(false);
 
-  const subscribe_ref: React.MutableRefObject<{ unsubscribe: () => void }> = useRef();
-
   useEffect(() => {
```

Comment:
Instead of using 2 seperate use effects can we use 1?

Reply:
The effects handle two seperate things. I believe it's better to seperate those two concerns in their own useEffect. React docs also suggests this: https://legacy.reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns

Reply:
Not exactly sure, I did now as you suggested. I ususally think of seperation of concerns when it comes with useeffect. But I think we can combine these two in one useeffect yes, done!

Reply:
Nevermind, I reverted the change and stick to the seperation of concerns (two useeffects). Tests fail now because the unsubscribe functionality gets called multiple times now. So I think seperation of concerns is definitely the more viable way to go.